lib/GenomeFileUtil/core/GenomeUtils.py

```diff
@@ -229,3 +229,152 @@ def check_full_contig_length_or_multi_strand_feature(feature, is_transpliced, co
+                if feature['parent_gene'] not in feature_id_sets_dict['features']:
+                    not_found_relationships['parent_gene'] = [feature['parent_gene']]
+                if "cds" in feature:
+                    if feature['cds'] not in feature_id_sets_dict['cdss']:
+                        not_found_relationships['cds'] = [feature['cds']]
+                break                       
+    elif feature_id in feature_id_sets_dict['non_coding_features']:
+        # NEED TO CHECK BOTH FEATURES AND NON_CODING_FEATURES FOR PARENT_GENE
+        # Children will only be NON_CODING_FEATURES
+        for feature in genome["non_coding_features"]:
+            if feature_id == feature["id"]:
+                # Do parent could be in either feature or non_coding_features (Only 1 parent)
+                if "parent_gene" in feature:
+                    if feature['parent_gene'] not in feature_id_sets_dict['features'] and \
+                    feature['parent_gene'] not in feature_id_sets_dict['non_coding_features']:
+                        not_found_relationships['parent_gene'] = [feature['parent_gene']]
+                if "children" in feature:
+                    not_found_children = list()
+                    for child in feature['children']:
+                        if child not in feature_id_sets_dict['non_coding_features']:
+                            not_found_children.append(child)
+                    if len(not_found_children) > 0:
+                        not_found_relationships['children'] = not_found_children
+                break    
+    else:    
+        # Raise an error the searched for feature does not exist in any of the 4 lists.
+        raise ValueError('Feature ID : ' + feature_id + ' was not found in the genome.')
+    return not_found_relationships
+
+
+def confirm_genomes_feature_relationships(genome):
+    ''' 
+    Confirms the relationships of all features in a genome
+    Note this is not a quick operation, should be used sparingly and not necessarily for every genome
+    Takes a genome and returns a dict with feature ids as the key and a dict of relationship type and missing features
+    NOTE THIS DOES NOT INSURE THAT RELATIONSHIPS ARE RECIPROCAL. JUST CHECKS THAT A FEATURE EXISTS FOR LISTED RELATIONSHIPS.
+    '''
+    features_with_relationships_not_found = dict()
+    feature_lists = ["features","cdss","mrnas","non_coding_features"]
+    # dict is the feature list and the key is the set of ids in that list.
+    feature_id_sets_dict = dict() 
+    for feature_list in feature_lists:
+        if feature_list in genome:
+            feature_id_sets_dict[feature_list] = make_id_set(genome[feature_list])
+        else:
+            feature_id_sets_dict[feature_list] = set()
+    for feature_list in feature_lists:
+        for feature in genome[feature_list]:
+            feature_relationship_dict = confirm_feature_relationships(genome, feature["id"], feature_id_sets_dict)
```

Comment:
Rather than passing the ID and genome, pass the actual feature and it's type so that you don't have to look through the array gain to find it.